### PR TITLE
use MiB units for memory per node

### PR DESCRIPTION
**Pull request by smoors**
fixes https://github.com/EESSI/test-suite/issues/150

- use MiB for `mem_per_node` in config files
- use MiB for `app_mem_req` argument of hook `req_memory_per_node`

### CODE REVIEW

File: eessi/testsuite/hooks.py [hunk 1]
```diff
@@ -396,59 +396,57 @@ def req_memory_per_node(test: rfm.RegressionTest, app_mem_req):
...
     # Fraction of CPU cores requested
     check_proc_attribute_defined(test, 'num_cpus')
     cpu_fraction = test.num_tasks_per_node * test.num_cpus_per_task / test.current_partition.processor.num_cpus
-    proportional_mem = cpu_fraction * test.current_partition.extras['mem_per_node']
+    proportional_mem = math.floor(cpu_fraction * test.current_partition.extras['mem_per_node'])
+    app_mem_req = math.ceil(app_mem_req)
 
     scheduler_name = test.current_partition.scheduler.registered_name
     if scheduler_name == 'slurm' or scheduler_name == 'squeue':
-        # SLURMs --mem defines memory per node, see https://slurm.schedmd.com/sbatch.html
```
[the remark this answers is not in the document]
Comment (smoors) [reply eessi/testsuite/hooks.py:431]: yeah, never trust when they megabytes. resistance to the new SI standard is very strong regarding bytes units.

**Comment by smoors** [comment]
@laraPPr i updated the `mem_per_node` values in the vsc_hortense config file according the slurm.conf file, which should have the correct values. if not, make sure to update your slurm.conf

**Comment by smoors** [comment]
@casparvl i just multiplied the `mem_per_node` values with 1024 for the clusters you have access to. you might want to update those to more accurate values

**Comment by laraPPr** [comment]
> @laraPPr i updated the `mem_per_node` values in the vsc_hortense config file according the slurm.conf file, which should have the correct values. if not, make sure to update your slurm.conf

You just converted to MiB? Than it should be okay tested it on saturday.

**Comment by smoors** [comment]
> > @laraPPr i updated the `mem_per_node` values in the vsc_hortense config file according the slurm.conf file, which should have the correct values. if not, make sure to update your slurm.conf
> 
> You just converted to MiB? Than it should be okay tested it on saturday.

i just copied what is in the slurm.conf, those values are already in MiB.

**Comment by laraPPr** [comment]
> > > @laraPPr i updated the `mem_per_node` values in the vsc_hortense config file according the slurm.conf file, which should have the correct values. if not, make sure to update your slurm.conf
> > 
> > 
> > You just converted to MiB? Than it should be okay tested it on saturday.
> 
> i just copied what is in the slurm.conf, those values are already in MiB.
 
Ah, Yes should be okay since I also got it from slurm.conf.

**Comment by smoors** [comment]
@laraPPr this is in your slurm.conf
```
$ scontrol show partition |grep -e PartitionName -e MaxMemPerNode    
PartitionName=cpu_milan
   DefMemPerCPU=1970 MaxMemPerNode=252160
PartitionName=cpu_rome
   DefMemPerCPU=1970 MaxMemPerNode=252160
PartitionName=cpu_rome_512
   DefMemPerCPU=3970 MaxMemPerNode=508160
PartitionName=cpu_rome_all
   DefMemPerCPU=1970 MaxMemPerNode=252160
PartitionName=debug_rome
   DefMemPerCPU=4992 MaxMemPerNode=239640
PartitionName=gpu_rome_a100
   DefMemPerCPU=5300 MaxMemPerNode=254400
PartitionName=gpu_rome_a100_40
   DefMemPerCPU=5300 MaxMemPerNode=254400
PartitionName=gpu_rome_a100_80
   DefMemPerCPU=10640 MaxMemPerNode=510720
````

File: eessi/testsuite/hooks.py [hunk 2]
```diff
@@ -396,59 +396,57 @@ def req_memory_per_node(test: rfm.RegressionTest, app_mem_req):
 
     Arguments:
     - test: the ReFrame test to which this hook should apply
-    - app_mem_req: the amount of memory this application needs (per node) in GiB
+    - app_mem_req: the amount of memory this application needs (per node) in MiB
 
     Example 1:
     - A system with 128 cores and 64 GiB per node.
```
Comment (casparvl) [inline eessi/testsuite/hooks.py:402]:
```suggestion
    - A system with 128 cores and 64,000 MiB per node.
```

File: eessi/testsuite/hooks.py [hunk 3]
```diff
@@ -396,59 +396,57 @@ def req_memory_per_node(test: rfm.RegressionTest, app_mem_req):
...
-    - The app_mem_req is 40 (GiB)
-    In this case, the test requests 50% of the CPUs. Thus, the proportional amount of memory is 32 GiB.
-    The app_mem_req is higher. Thus, 40GiB (per node) is requested from the batch scheduler.
+    - The app_mem_req is 40,000 (MiB)
+    In this case, the test requests 50% of the CPUs. Thus, the proportional amount of memory is 32,000 MiB.
+    The app_mem_req is higher. Thus, 40,000 MiB (per node) is requested from the batch scheduler.
 
     Example 2:
-    - A system with 128 cores per node, 128 GiB mem per node is used.
+    - A system with 128 cores per node, 128,000 MiB mem per node is used.
```
Comment (casparvl) [inline eessi/testsuite/hooks.py:409]:
```suggestion
    - A system with 128 cores per node, 128,000 MiB mem per node.
```

Comment (casparvl) [reply eessi/testsuite/hooks.py:409]: "Is used" was not supposed to be there (that was my bad actually, but lets fix it)

**Review by casparvl — APPROVED** [review]
Looks good to me. I'll make a PR later to check the number for the configs once more... For now, lets get this in since the test step in EESSI's `software-layer` is currently broken ;-)

**Pull request merged**